src/edu/washington/escience/myria/storage/TupleBuffer.java

```diff
@@ -361,6 +361,5 @@ public final void put(
       final int destColumn, final ReadableColumn sourceColumn, final int sourceRow) {
     checkPutIndex(destColumn);
     TupleUtils.copyValue(sourceColumn, sourceRow, this, destColumn);
-    columnPut(destColumn);
```

Comment:
It's in `TupleUtils.copyValue`, this function calls `putXXX()`s which call `columnPut()`